pkg/cloud/gcpprovider.go

```diff
@@ -1365,17 +1365,24 @@ func (gcp *gcpKey) GPUType() string {
 
 // GetKey maps node labels to information needed to retrieve pricing data
 func (gcp *gcpKey) Features() string {
+	var instanceType string
 	it, _ := util.GetInstanceType(gcp.Labels)
-	instanceType := strings.ToLower(strings.Join(strings.Split(it, "-")[:2], ""))
-	if instanceType == "n1highmem" || instanceType == "n1highcpu" {
-		instanceType = "n1standard" // These are priced the same. TODO: support n1ultrahighmem
-	} else if instanceType == "n2highmem" || instanceType == "n2highcpu" {
-		instanceType = "n2standard"
-	} else if instanceType == "e2highmem" || instanceType == "e2highcpu" {
-		instanceType = "e2standard"
-	} else if strings.HasPrefix(instanceType, "custom") {
-		instanceType = "custom" // The suffix of custom does not matter
+	if it == "" {
```

Comment:
@nikovacevic I think I'd like to leave it as it is to avoid the potential panic that may come from util.GetInstanceType returning an empty string (which occurs when the node type label is missing).

Reply:
Fair enough! Just to be clear: in that case (getting an empty string) the `len(...) < 2` check would also avoid panic, I believe. We'll just have to be aware that, with this fix, if GCP changes naming conventions we may still panic. (e.g. we will now panic if there is an instance type with no hyphens)